validator_client/src/duties_service.rs

```diff
@@ -488,6 +490,29 @@ async fn poll_validator_indices<T: SlotClock + 'static, E: EthSpec>(
             .is_some();
 
         if !is_known {
+            let current_slot_opt = duties_service.slot_clock.now();
+
+            if let Some(current_slot) = current_slot_opt {
+                let is_first_slot_of_epoch = {
+                    let current_epoch_first_slot = current_slot
+                        .epoch(E::slots_per_epoch())
+                        .start_slot(E::slots_per_epoch());
+                    current_slot == current_epoch_first_slot
+                };
```

Comment:
This could be a bit simpler as:

```suggestion
               let is_first_slot_of_epoch = current_slot % E::slots_per_epoch() == 0;
```

That's usually how we do it

Reply:
Nice, thanks!